crap-code.kt

```diff
@@ -0,0 +1,26 @@
+package badcode.example
+class PerformanceFile {
+    val largeList = List(1000000) { it }
+    // Inefficient way to calculate the sum of elements
+    var sum = 0
+    fun calculateSum() {
+        for (i in 0 until largeList.size) {
+            sum += largeList[i]
+        }
+        // typo in println() function
+        prentln("Sum: $sum")
+        // Unnecessary object creation inside a loop
+        for (i in 0 until 100000) {
+            val tempObject = Any() // Creates a new object on each iteration
+        }
+        // Inefficient string concatenation
+        var longString = ""
+        for (i in 0 until 10000) {
+            longString += "a" // Creates a new string object on each concatenation
+        }
+        println("Long String Length: ${longString.length}")
+        // Blocking the main thread with a long-running operation
+        Thread.sleep(5000) // Simulates a blocking operation
+        println("Operation complete")
+    }
```

Comment:
Avoid blocking the main thread with `Thread.sleep`; consider using asynchronous processing.
```suggestion
import kotlinx.coroutines.*

fun calculateSum() {
    GlobalScope.launch {
        for (i in 0 until largeList.size) {
            sum += largeList[i]
        }
        prentln("Sum: $sum")
        for (i in 0 until 100000) {
            val tempObject = Any()
        }
        var longString = ""
        for (i in 0 until 10000) {
            longString += "a"
        }
        println("Long String Length: ${longString.length}")
        delay(5000)
        println("Operation complete")
    }
}
```